src/core/analysis-context.ts

```diff
@@ -496,4 +487,28 @@ export class AnalysisContext {
     return this.resolver.canResolve(url) ? this.resolver.resolve(url) :
                                            (url as any as ResolvedUrl);
   }
+
+  /**
+   * A warning for a weird situation that should not obtain.
```

Comment:
"should not obtain" -> "should not happen"

Reply:
done